### PR TITLE
Device Update GA preview for Java

### PR DESCRIPTION
**Pull request by dpokluda**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
checkstyle error:
```
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java:7:8: Unused import - java.time.OffsetDateTime. [UnusedImports]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java:8:8: Unused import - java.time.ZoneOffset. [UnusedImports]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:54:17: Name 'testGetDevice_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:81:17: Name 'testGetGroup_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:101:17: Name 'testGetDeviceClasses_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:121:17: Name 'testGetBestUpdatesForGroup_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:143:17: Name 'testGetDeploymentsForGroup_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java:63:17: Name 'testGetNames_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java:85:17: Name 'testGetVersions_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java:125:17: Name 'testGetFiles_NotFound' must match pattern '(?=^[a-z][a-zA-Z0-9]*$)(?!^(?i)builder$)'. [MethodName]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java:8:8: Unused import - com.azure.core.http.rest.PagedFlux. [UnusedImports]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java:13:8: Unused import - com.fasterxml.jackson.databind.ObjectMapper. [UnusedImports]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetUpdateSample.java:8:8: Unused import - com.azure.core.http.rest.PagedFlux. [UnusedImports]
[ERROR] /mnt/vss/_work/1/s/sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetUpdateSample.java:13:8: Unused import - com.fasterxml.jackson.databind.ObjectMapper. [UnusedImports]
```

You can run locally `mvn checkstyle:check`

**Comment by dpokluda** [comment]
Fixed the check-style errors.

File: sdk/deviceupdate/azure-iot-deviceupdate/README.md [hunk 1]
```diff
@@ -54,17 +54,6 @@ All Device Update for IoT Hub service operations will throw a ErrorResponseExcep
 
 For example, if you use the `getUpdateWithResponse` operation and the model you are looking for doesn't exist, you can catch that specific HttpStatusCode to decide the operation that follows in that case.
 
-``` java com.azure.iot.deviceupdate.DeviceUpdateAsyncClient.notfound
```
Comment (haolingdong-msft) [inline sdk/deviceupdate/azure-iot-deviceupdate/README.md:57]: Why delete those codesnippet?
This document shows how to add codesnippet: https://github.com/Azure/azure-sdk-tools/blob/main/packages/java-packages/codesnippet-maven-plugin/README.md

File: sdk/deviceupdate/azure-iot-deviceupdate/src/samples/README.md [hunk 2]
```diff
@@ -1,39 +1,28 @@
...
+| DEVICEUPDATE_UPDATE_NAME     | vacuum               |
+| DEVICEUPDATE_UPDATE_VERSION  | 2022.401.504.6       |
+| DEVICEUPDATE_DEVICE          | dpokluda-test        |
 
 ## Creating Client
 
 You need to use AzureDeviceUpdateClientBuilder to create a proper valid AzureDeviceUpdateClient. This main client will
-let you access the other clients - Updates and Management. In the sample `ListDevicesSample`, it creates `DeviceManagementAsyncClient`.
-
-```java com.azure.iot.deviceupdate.DeviceManagementAsyncClient.instantiate
```
Comment (haolingdong-msft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/samples/README.md:26]: Why delete this code snippet?

**Comment by haolingdong-msft** [comment]
Hi @dpokluda, I use latest autorest.java to generate deviceupdate again. This is pr: https://github.com/Azure/azure-sdk-for-java/pull/29365. This is the commit of my code change: https://github.com/Azure/azure-sdk-for-java/pull/29365/commits/399b443e33a643bb0904ea5b6a6f3046f9e6b34e
Could you please decide if you want this?

File: sdk/deviceupdate/azure-iot-deviceupdate/README.md (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/deviceupdate/azure-iot-deviceupdate/README.md:57]: SDK would prefer we had as least one sample in README.md, so customer know how to authenticate and call API.

Please see above doc on how to sync your sample java to the README.md

Alternative doc https://github.com/Azure/azure-sdk-for-java/wiki/Protocol-Methods-Quickstart-with-AutoRest#samples

File: sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java [hunk 3]
```diff
@@ -5,28 +5,31 @@
...
             .credential(new DefaultAzureCredentialBuilder().build())
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .buildAsyncClient();
 
-        // BEGIN: com.azure.iot.deviceupdate.DeviceUpdateAsyncClient.notfound
         try {
-            client.getUpdateWithResponse("foo", "bar", "0.0.0.1", null).block();
+            Response<BinaryData> response = client.getDeviceWithResponse(
+                Configuration.getGlobalConfiguration().get("DEVICEUPDATE_DEVICE"),
+                null).block();
```
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java:25]: The common usage on sync API is to create `DeviceManagementClient` via `.buildClient()`.

And then API call does not need to have `.block()` or `.toIterable()` to convert async response to sync response.

File: sdk/deviceupdate/azure-iot-deviceupdate/README.md (hunk 1 above)
Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/README.md:57]: Sorry, misunderstood the purpose of those. I brought them back. Hope it helps.

File: sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java (hunk 3 above)
Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/samples/java/com/azure/iot/deviceupdate/GetDeviceSample.java]: Got it. Sorry I am not Java person, I am just mostly doing copy/paste. :-(

**Comment by dpokluda** [comment]
Adding sample to README.md.

File: sdk/deviceupdate/azure-iot-deviceupdate/src/samples/README.md (hunk 2 above)
Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/samples/README.md:26]: I will admit that after I have updated the samples, the compilation failed with missing snippet. I didn't know how to fix it and removing this from the README was the only solution I came up with. It is fixed now properly. My bad, sorry.

**Comment by dpokluda** [comment]
@haolingdong-msft, I have used the following command to update the client library using the latest autorest.java extension (version 4.0.61):

```
 autorest --version=3.8.1 --use=@autorest/java@4.0.61 --java --java-sdks-folder=E:\Sdk\azure-sdk-for-java E:\Sdk\azure-sdk-for-java\sdk\deviceupdate\azure-iot-deviceupdate\swagger\README.md
```

**Comment by haolingdong-msft** [comment]
> @haolingdong-msft, I have used the following command to update the client library using the latest autorest.java extension (version 4.0.61):
> 
> ```
>  autorest --version=3.8.1 --use=@autorest/java@4.0.61 --java --java-sdks-folder=E:\Sdk\azure-sdk-for-java E:\Sdk\azure-sdk-for-java\sdk\deviceupdate\azure-iot-deviceupdate\swagger\README.md
> ```

Thanks David, looks good for me.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
We need to update CHANGELOG.md before release, 'Unreleased' will be replaced as release date. Please update 'Features Added' and 'Bugs Fixed' part.

## 1.0.0-beta.3 (Unreleased)

### Features Added

### Breaking Changes

- Added `DeviceManagementClientBuilder` to create `DeviceManagementClient` instance via `DeviceManagementClientBuilder.buildClient()`.
- Modified `DeviceUpdateClientBuilder` to create `DeviceUpdateClient` instance via `DeviceUpdateClientBuilder.buildClient()`.
- Exposed required query and header parameters to method signature.

### Bugs Fixed

### Other Changes

#### Dependency Updates
- Upgraded `azure-core` to `1.29.1`.
- Upgraded `azure-core-http-netty` to `1.12.2`.

**Comment by weidongxu-microsoft** [comment]
comment on autorest.java bug in apiview fixed in @autorest/java@4.0.62

File: sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java [hunk 4]
```diff
@@ -41,39 +42,117 @@ private DeviceManagementAsyncClient createClient() {
...
+        try {
+            client.getDeviceWithResponse("foo", null).block();
+            fail("NotFound response expected");
+        } catch (HttpResponseException e) {
+            assertEquals(404, e.getResponse().getStatusCode());
+        }
+    }
+
+    // Temporary disabled because the service doesn't properly handle this method yet
+    // @Test
```
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientTests.java:65]: nit, in Java Junit, usually do this to disable a test.
```java
    @Test
    @Disabled
    public void testGetGroups() {
```

**Comment by weidongxu-microsoft** [comment]
I should be able to merge if apiview is approved by Srikanta.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

**Pull request merged**